modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerImpl.java

```diff
@@ -1080,6 +1080,7 @@ private boolean isAuthorized(final String mediaPackageId, final String action) {
                 .collect(Collectors.toList());
         return getDatabase().selectProperties(mediaPackageId, SECURITY_NAMESPACE).parallelStream()
                 .map(p -> p.getId().getName())
+                .filter(p -> p.endsWith(action))
                 .anyMatch(p -> roles.parallelStream().anyMatch(r -> r.equals(p)));
```

Comment:
Wouldn't it make sense to additionally remove the second parallel stream? I guess most of the time it won't even run in parallel because the default ForkJoinPool does not have that many threads.

```suggestion
                .anyMatch(p -> roles.stream().anyMatch(r -> r.equals(p)));
```

Reply:
No idea tbh. Maybe somebody else with more knowledge can chime in here?

Reply:
IMO removing it probably makes sense, but won't really matter perf wise.  How many roles is any one MP likely to have?  Running a regex against the full list sequentially is going to be as fast, if not faster than splitting it up into N substreams and doing the same thing against those in parallel.  But in the odd case where there are a truly astounding number of roles, is it *really* going to be that much faster to split them up?  I somehow doubt it.

In short, time(splitting substreams + attendant baggage + N * subsearch) > time(search).

Reply:
>  How many roles is any one MP likely to have?

You haven't seen what I've seen. ;)

But then I guess let's do this as well.